sdk/src/types/block/output/unlock_condition/mod.rs

```diff
@@ -355,6 +374,53 @@ pub(crate) fn verify_allowed_unlock_conditions(
     Ok(())
 }
 
+pub(crate) fn verify_restricted_addresses(
+    unlock_conditions: &UnlockConditions,
+    output_kind: u8,
+    native_token: Option<&NativeTokenFeature>,
+    mana: u64,
+) -> Result<(), Error> {
+    let addresses = unlock_conditions.restricted_addresses();
+
+    for address in addresses {
+        if native_token.is_some() && !address.has_capability(AddressCapabilityFlag::OutputsWithNativeTokens) {
+            return Err(Error::RestrictedAddressCapability);
```

Comment:
Should we return the specific capability that is restricted? Looks annoying to debug without it

Reply:
Sure, done